internal/dao/dp.go

```diff
@@ -211,6 +211,42 @@ func (d *Deployment) Scan(ctx context.Context, gvr, fqn string, wait bool) (Refs
 	return refs, nil
 }
 
+func (d *Deployment) GetPodSpec(path string) (*v1.PodSpec, error) {
+	dp, err := d.Load(d.Factory, path)
+	if err != nil {
+		return nil, err
+	}
+	podSpec := dp.Spec.Template.Spec
+	return &podSpec, nil
+}
+
+func (d *Deployment) SetImages(ctx context.Context, path string, containersPatch map[string]string, initContainersPatch map[string]string) error {
+	ns, n := client.Namespaced(path)
+	auth, err := d.Client().CanI(ns, "apps/v1/deployments", []string{client.PatchVerb})
+	if err != nil {
+		return err
+	}
+	if !auth {
+		return fmt.Errorf("user is not authorized to patch a deployment")
+	}
+	jsonPatch, err := GetTemplateJsonPatch(containersPatch, initContainersPatch)
+	if err != nil {
+		return err
+	}
+	dial, err := d.Client().Dial()
+	if err != nil {
+		return err
+	}
+	_, err = dial.AppsV1().Deployments(ns).Patch(
+		ctx,
+		n,
+		types.StrategicMergePatchType,
+		[]byte(jsonPatch),
```

Comment:
@Ameausoone This always ends up requiring []byte to might are well return the []byte from GetTemplateJsonPath vs returning a string

Reply:
of course